greenlight/lib/api/prefs.py

```diff
@@ -2,109 +2,174 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
+from marionette.errors import MarionetteException
+
 from ..base import BaseLib
 
 
-class DefaultPrefBranch(BaseLib):
+class Preferences(BaseLib):
     archive = {}
 
-    @classmethod
-    def _cast(cls, value):
-        """
-        Interpolate a preference from a string.
+    def clear_user_pref(self, pref_name):
+        """Clear a user set preference.
 
-        - integers will get cast to integers
-        - true/false will get cast to True/False
-        - anything enclosed in single quotes will be treated as a string with
-          the ''s removed from both sides
+        :param pref_name: The preference to clear
         """
+        with self.marionette.using_context('chrome'):
+            return self.marionette.execute_script("""
+              Cu.import("resource://gre/modules/Services.jsm");
+              let prefBranch = Services.prefs.QueryInterface(Ci.nsIPrefBranch);
 
-        if not isinstance(value, basestring):
-            return value  # no op
-        quote = "'"
-        if value == 'true':
-            return True
-        if value == 'false':
-            return False
-        try:
-            return int(value)
-        except ValueError:
-            pass
-        if value.startswith(quote) and value.endswith(quote):
-            value = value[1:-1]
-        return value
+              let pref_name = arguments[0];
+
+              if (prefBranch.prefHasUserValue(pref_name)) {
+                prefBranch.clearUserPref(pref_name);
+                return true;
+              }
+              else {
+                return false;
```

Comment:
Ah, right. I was just thinking because clearUserPref doesn't hurt when one isn't set, this could just be "prefBranch.clearUserPref(pref_name); return prefHasUserValue(pref_name);", so I started to think this is really a separate matter. Your call.